### ISSUE TITLE
[EN][PVP] Diminishing returns

### ISSUE THREAD

**Issue by brokenshard** [Atlantiss/BugTracker#9528]
Description: Diminshing returns duration is lower than it should be(after a crowd control effect ends, the diminishing return for that type of crowd control lasts for 18sec but here it lasts for 15secs
https://www.youtube.com/watch?v=lihXxekzHdM
u can see in the video that after a crowd cc ends the dr starts and lasts for 18secs
Current behaviour: drs(diminishing returns) last for 15sec only
Expected behaviour: drs should last for 18secs
Steps to reproduce the problem:
1. use any type of crowd control on a target
2. as soon as the cc ends start the stopwatch
3. wait untill 15secs have passed then use the same type of  crowd control again for example(fear/stun etc...)

**Comment by brokenshard** [comment]
its 18 on retail what do u mean

**Comment by blinkmaster** [comment]
http://wowwiki.wikia.com/wiki/Diminishing_returns
Between 15 - 20 sec range

**Comment by brokenshard** [comment]
it legit says in the video that its 18 and the article that u linked is from wotlk so its out of date

**Comment by fluessig** [comment]
"It should be noted that the server does not reset diminishing returns after exactly 15 seconds. Instead, it checks every 5 seconds if there are any diminishing returns that should be reset because the last spell in their category was cast on the target more than 15 seconds ago. As a result, a particular diminishing return category may take anywhere between 15 and 20 seconds to reset. Most addons guess that a DR category has reset after 18 seconds by default."

just read what it says
who would want a random timer for DR :(
id rather have it be 20 or 15
it was clearly intended to be 15 seconds. but blizzard couldnt handle to ask for this info on all theii server more often than every 5 seconds

**Comment by brokenshard** [comment]
this is the right article it says 18sec https://wow.gamepedia.com/Diminishing_returns

**Comment by lamaATS** [comment]
First look into this issue and what I found in TC [here](https://github.com/TrinityCore/TrinityCore/pull/13907) DR was changes from 15s to 18s in WoD.

**Comment by kadi2** [comment]
Gladius and most addons are adjusted to 18 seconds timer. You'd need to be very unlucky to have its duration longer than 18 seconds, but that was common knowledge as most sources provide that information.

<<It should be noted that the server does not reset diminishing returns after exactly 15 seconds. Instead, it checks every 5 seconds if there are any diminishing returns that should be reset because the last spell in their category was cast on the target more than 15 seconds ago. As a result, a particular diminishing return category may take anywhere between 15 and 20 seconds to reset. Most addons guess that a DR category has reset after 18 seconds by default.'

http://www.arenajunkies.com/topic/223931-diminishing-returns/

**Comment by lamaATS** [comment]
[Here](https://us.battle.net/forums/en/wow/topic/16529192789) you can see that DR was changed to 18s in patch 6.1 and if I'm reading it correctly before this it was random between 15 and 20 seconds. Currently on Atlantiss we have exactly 15 seconds (since years).

**Comment by Aival94** [comment]
Not sure if it helps but i found these: https://us.battle.net/forums/en/wow/topic/10195910192 (is from mop which state that it was flat 15 so if there was no change before that then in cata should be same)
Furthermore wowpedia changed to 18second after 2016. During cata the wowpedia post was that: https://wow.gamepedia.com/index.php?title=Diminishing_returns&oldid=2977495

**Comment by tomatosz** [comment]
Brokenshard the wowwiki site may have been out of date, but yours was too new. Always check in history, this is the last update on gamepedia before MoP:
https://wow.gamepedia.com/index.php?title=Diminishing_returns&oldid=2977495
And this mentions 15sec.

**Comment by lamaATS** [comment]
So we have two possible cases:
* const 15 seconds
* some kind of random between 15 and 20 seconds

18 seconds was a change in patch 6.1. Currently on server we have 15 seconds.
My suggestion is to change mechanic like it's explained in @kadi2 comment:

> Instead, it checks every 5 seconds if there are any diminishing returns that should be reset because the last spell in their category was cast on the target more than 15 seconds ago.

As we can see in BluePost:

> In patch 6.0.3, due to the **`method`** used to update diminishing returns, that window could last anywhere from 15 to 20 seconds.

For me described mechanic makes sense.

Waiting for more comments with Agree or Disagree (explanation will be useful).

**Comment by blinkmaster** [comment]
Id rather have constant dr timer than ats rng =)

**Comment by lamaATS** [comment]
Random Number Generator will not be used for it. I currently implemented on my local server.

**Comment by lamaATS** [comment]
So constant it's not blizzlike but players want it?

**Comment by fluessig** [comment]
could maybe do a poll on forums? about this

**Comment by lamaATS** [comment]
I probably know result now, it will be constant 15 seconds because people will not care about blizzlike but what is better for them and 15 seconds is always shorten then 15-20 seconds so for me it doesn't make sense.

**Comment by Aival94** [comment]
I think best should be retail way as most addons that many PVPers use for DR are on 18sec which means if you put it on 15sec will be strange to always have 15sec instead. Furthermore retail value feel better and more hard to control which will make PVP a little more challenging to those that want to use their ccs. If it is constant 15sec is like you give them an easy way and i think that is not the reason of playing PVP.

**Comment by lamaATS** [comment]
Can someone here list some Addons which support DRs? I will check code how it's working inside maybe this will tell something more.

**Comment by fluessig** [comment]
well if we look at cc in the game it can get pretty clear that 15 second was the intended DR time

dragons breath 
20 second cd
5 sec duration
15 sec downtime
so u can actually use Dragons breath right on cd and not dr with yourself

priest fear with shadow talents and gloves pvp bonus 
23 second cd 
8 sec duration 
makes 15 second downtime exactly

maybe putting it at a constant 18 seconds is the best compromise

**Comment by kadi2** [comment]
Gladius and probably all other addons have a constant variable that's equal to 18 seconds.

**Comment by Soulywood** [comment]
I agree to put it at 18secs even if it's not 100% blizzlike at least it would be better than the current state because 15secs is just not representative of what you could find on retail

**Comment by Aival94** [comment]
@lamaATS as far as i saw there were no released version of addon with diminishing returns. All were beta,alpha or were released during 4.2. If you still want to check some of them here is a list:

[DiminishingReturns](https://mods.curse.com/addons/wow/diminishingreturns#t1:other-downloads) - File: DiminishingReturns\libs\DRData-1.0.lua
[Gladius](https://wow.curseforge.com/projects/gladius/files?filter-game-version=2020709689%3A181) - File: Gladius\Libs\DRData-1.0 (Note: "-- As of 6.1, this is always 18 seconds, and no longer has a range between 15 and 20 seconds." and "-- Knockbacks are a special case"
[TellMeWhen](https://wow.curseforge.com/projects/tellmewhen/files?filter-game-version=2020709689:205) - File: TellMeWhen\Lib\DRData-1.0

**Comment by PvPbugreporter** [comment]
http://www.skill-capped.com/guides/shared_drs.php
http://wowwiki.wikia.com/wiki/Diminishing_returns
"Within 15 seconds of the *End* of the effect. Here they start and go from within 15 seconds of the beginning, hence the bug and the short length."

DR= 15 seconds. If i use an 8 sec fear, as soon as it ends the 15 seconds start. So from the beginning of the fear it will be 23 seconds until dr falls off. If i use a 4 second cheap shot 15 seconds after the duration of the cs *ends* the dr will fall off. From the time you use the cheap shot it will be 19 seconds before it can be used again. (This is how it should work and this is how it was on retail)

On Atlantiss the dr is the correct time but if I use cheap shot here (Or any cc in the game) the timer starts when the spell takes effect meaning 15 seconds after i use the spell the dr falls off rather than the blizzlike coding where it would fall off 15 sec after the *end* of the spells duration.

This is simple to understand, please do not break it more by reporting it should be a different amount of time. Atlantiss DR times are 100% correct, the problem with why they seem short is because the time starts at the incorrect time. Hopefully this issue will be fixed the proper way.
